app/js/enhancedDate.js

```diff
@@ -1,5 +1,71 @@
+var enhancedDate = (function() {
+	var dateIsSet;
+	var enhancedDateObject = {
+		setDate: setDate,
+		getDate: getDate,
+		getDayName: getDayName,
+		getMonthName: getMonthName,
+		isFuture: isFuture,
+		isToday: isToday
+	};
+	return enhancedDateObject;
+
+
+	function setDate(date) {
+		if (!date) {dateIsSet = new Date(); return;}
+		if (!(date instanceof Date)) {console.warn('sorry, not a date'); return;}
+		dateIsSet = new Date(date);
+	};
+
+	function getDate() {
+		isDateSet();
+		return dateIsSet;
+	}
+
+	function getDayName() {
+		isDateSet();
+		var dayName = dateIsSet.getDay();
+		var dayOfWeekArray = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
```

Comment:
This is the right way to do this, but putting the arrays of day names in the module but outside of the method would be more efficient. Each time getDayName is called, it will re-create this array.